server/game/GameActions/ConditionalGameAction.ts

```diff
@@ -0,0 +1,66 @@
+import { GameAction, GameActionProperties } from './GameAction';
+import AbilityContext = require('../AbilityContext');
+
+export interface ConditionalActionProperties extends GameActionProperties {
+    gameAction: GameAction;
+    condition: boolean | ((context: AbilityContext) => boolean);
+    else?: GameAction;
+}
+
+export class ConditionalGameAction extends GameAction {
+    effect = 'do a conditional action';
+
+    constructor(properties: ConditionalActionProperties | ((context: AbilityContext) => ConditionalActionProperties)) {
+        super(properties);
+    }
+
+    getProperties(context: AbilityContext, additionalProperties = {}): ConditionalActionProperties {
+        let properties = super.getProperties(context, additionalProperties) as ConditionalActionProperties;
+        properties.condition = properties.condition || true;
+        return properties;
+    }
+
+    meetsCondition(context: AbilityContext, properties: ConditionalActionProperties): boolean {
+        if(typeof properties.condition === 'function') {
+            return properties.condition(context);
+        }
+        return properties.condition as boolean;
+    }
+
+    hasLegalTarget(context: AbilityContext, additionalProperties = {}): boolean {
+        let properties = this.getProperties(context, additionalProperties);
+        if(this.meetsCondition(context, properties)) {
+            return properties.gameAction.hasLegalTarget(context, additionalProperties) || properties.else.hasLegalTarget(context, additionalProperties);
```

Comment:
This will throw an error if gameAction isn't legal and properties.else isn't defined